org_status/org_status.py

```diff
@@ -148,55 +175,78 @@ def main():
 
     all_repositories = []
 
-    for Host, org in generate_fetch_jobs(args.orgs):
-        token = None
+    if args.use_repo_list:
+        repos = decode_repo_list(args.use_repo_list, args.format, styled)
 
-        if not args.skip_host_checks:
-            try:
-                if not Host.get_host_status():
-                    print(styled(f'{Host.HostName} is currently down', 'red'))
-                    continue
-                else:
-                    hosts_only_print(f'{Host.HostName} is up')
-            except NotImplementedError:
-                verbose(
-                    f'{Host.HostName} does not support checking host status')
+        for Host, org in generate_fetch_jobs(args.orgs):
+            repo_names = []
+            repo_urls = []
+            for repo in repos:
```

Comment:
This should not be done this way. We will have two sections of code essentially doing the same thing and this makes it even tougher to manage. We want `sync=False` and `sync=True` (i.e using a repo list or fetching a fresh one) to behave as much similarly possible and reuse as much code as possible.

Please drop all this redundant code and find a way to  hack in `for Host, org in generate_fetch_jobs(args.orgs)` to make it work.